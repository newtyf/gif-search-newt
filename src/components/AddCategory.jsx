import React, { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const handleChange = ({ target }) => {
    setInputValue(target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length < 1) {
      return alert("la categoria debe ser mas larga");
    }
    // setCategories((catergories) => [inputValue, ...catergories]);
    onNewCategory(inputValue.trim())
    setInputValue("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={handleChange}
      />
    </form>
  );
};
