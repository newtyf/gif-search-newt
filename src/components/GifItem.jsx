import React from "react";
import PropTypes from 'prop-types'

export const GifItem = ({ title, imgUrl }) => {
  return (
    <div className="card">
      <img src={imgUrl} alt={title} />
      <p aria-label="title-gif" >{title.length < 2 ? "No hay titulo" : title}</p>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  imgUrl: PropTypes.string.isRequired
}