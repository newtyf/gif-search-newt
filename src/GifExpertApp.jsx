import React, { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch"]);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory))
      return alert("La categoria ingresada ya existe");

    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onNewCategory={onAddCategory} />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
