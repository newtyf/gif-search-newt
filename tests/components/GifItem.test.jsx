import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("Pruebas en <GifItem />", () => {
  const title = "One Punch Man Gif";
  const url =
    "https://media3.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=ea146d384s28crehs3279gymn9x2qn3vzasxbq9pw0asjjfi&rid=giphy.gif&ct=g";

  test("debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} imgUrl={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={title} imgUrl={url} />);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={title} imgUrl={url} />);
    expect(screen.getByText(title)).toBeTruthy()
  });
});