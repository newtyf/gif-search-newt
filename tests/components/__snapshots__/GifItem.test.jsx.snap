// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch Man Gif"
      src="https://media3.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=ea146d384s28crehs3279gymn9x2qn3vzasxbq9pw0asjjfi&rid=giphy.gif&ct=g"
    />
    <p
      aria-label="title-gif"
    >
      One Punch Man Gif
    </p>
  </div>
</div>
`;
